### PR TITLE
fix(app): Fix issue when file folder does not exist

### LINKED ISSUE #18: exorcist fails silently when the output dir does not exist
I'd prefer exorcist do a mkdir -p to ensure the output directory exists, but it should at least throw an error if it doesn't.

### PR DESCRIPTION
**Pull request by thaiat**
Fixes #18

### CODE REVIEW

**Comment by marnusw** [comment]
+1

I wasted about an hour trying to figure out why the source map wasn't generated. Turns out it's because the target folder was cleaned away and not present when the source map was extracted.

**Comment by thaiat** [comment]
@thlorenz have you had a chance to look at the PR... sadely i don't see much activity on the repo lately

**Comment by devinus** [comment]
Holy shit, this is much needed. So much time wasted...

File: package.json [hunk 1]
```diff
@@ -21,6 +21,7 @@
   "homepage": "https://github.com/thlorenz/exorcist",
   "dependencies": {
     "minimist": "0.0.5",
+    "mkdirp": "^0.5.1",
```
Comment (thlorenz) [inline package.json]: Please stick to `~` ranges.

File: index.js [hunk 2]
```diff
@@ -60,7 +61,9 @@ function exorcist(file, url, root, base) {
       );
       return write(src.source);
     }
-
+    // make sure the file folder exists
+    mkdirp.sync(path.dirname(file));
```
Comment (thlorenz) [inline index.js]: Can't have sync calls in a lib (people may use it as such even if main use case is bin tool).
Please find a way to do this async.

**Comment by thlorenz** [comment]
Sorry was super busy.
Please fix the issues I pointed out and add a test that fails before your feature add and passes after.

Thanks.

**Comment by abritinthebay** [comment]
@thaiat - if you're still interested in adding this can you make those fixes that thlorenz pointed out (and merge, as you have conflicts) ...?

**Pull request closed without merging**